source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -205,6 +209,29 @@ void TransformationFilter::addEncoderData(Buffer::Instance &data) {
   encoder_callbacks_->addEncodedData(data, false);
 }
 
+void TransformationFilter::transformOnStreamCompletion() {
+  if (on_stream_completion_transformation_ == nullptr) {
+    return;
+  }
+
+  Http::StreamFilterCallbacks *on_complete_callbacks{};
```

Comment:
Do we not have access to the actual stream callbacks? we may need those to mask dynamic metadata?

Reply:
There is an encoder and decoder callback. Since we were pass the encode/decode stage I just created a new callback. Do you think those callbacks should be passed in instead?